src/components/img.tsx

```diff
@@ -0,0 +1,94 @@
+// ----- Imports ----- //
+
+import React, { FC } from 'react';
+import { SerializedStyles, css } from '@emotion/core';
+import { Option, OptionKind, withDefault } from '@guardian/types/option';
+import { Format, Design } from '@guardian/types/Format';
+import { neutral } from '@guardian/src-foundations/palette';
+
+import { Image, Role } from 'image';
+import { darkModeCss } from 'lib';
+
+
+// ----- Functions ----- //
+
+const backgroundColour = (format: Format): string => {
+    switch (format.design) {
+        case Design.Media:
+            return neutral[20];
+        case Design.Comment:
+            return neutral[86];
+        default:
+            return neutral[97];
+    }
+};
+
+const getLightboxClassName = (
+    imageWidth: number,
+    className: Option<string>,
+): string | undefined => {
+    if (imageWidth > 620 && className.kind === OptionKind.Some) {
+        return className.value;
+    }
+
+    return undefined;
+}
+
+
+// ----- Component ----- //
+
+interface Props {
+    image: Image;
+    sizes: string;
+    className: Option<SerializedStyles>;
+    format: Format;
+    supportsDarkMode: boolean;
+    lightboxClassName: Option<string>;
+}
+
+const styles = (
+    role: Role,
+    format: Format,
+    supportsDarkMode: boolean,
+): SerializedStyles => {
+    switch (role) {
+        case Role.Thumbnail:
+            return css`color: ${neutral[60]};`;
+        default:
+            return css`
+                background-color: ${backgroundColour(format)};
+                color: ${neutral[60]};
+                ${darkModeCss(supportsDarkMode)`background-color: ${neutral[20]};`}
+            `;
+    }   
+}
+
+const Img: FC<Props> = ({ image, sizes, className, format, supportsDarkMode, lightboxClassName }) =>
+    <picture>
+        <source
+            sizes={sizes}
+            srcSet={image.dpr2Srcset}
+            media="(-webkit-min-device-pixel-ratio: 1.25), (min-resolution: 120dpi)"
+        />
+        <source
+            sizes={sizes}
+            srcSet={image.srcset}
+        />
+        <img
+            src={image.src}
+            alt={withDefault('')(image.alt)}
+            className={getLightboxClassName(image.width, lightboxClassName)}
+            css={[
+                styles(image.role, format, supportsDarkMode),
+                withDefault<SerializedStyles | undefined>(undefined)(className),
+            ]}
+            data-ratio={image.height / image.width}
```

Comment:
Might be worth implementing some `padding-top: ${100/ratio}%` ?

https://css-tricks.com/aspect-ratio-boxes/

Reply:
Interesting. Is there a reason that's preferable to doing:

```ts
height: ${100/ratio}%;
```

or even:

```ts
height: calc(100% * ${ratio});
```

?

Also worth noting that this component may typically be used by a more specific image component that's more prescriptive about the styles, e.g. [BodyImage](https://github.com/guardian/apps-rendering/blob/d1e6268b9ddcc28e0458be6c248ca7fda404f474/src/components/bodyImage.tsx#L25), which I plan to implement next. That said, it may be preferable to always set the `height` on the `Img` component using percentages, because it's unlikely that we'll want the ratio to change.

Reply:
I've pushed a change to specify height for now.

Reply:
Yes, setting width and height is ideal, but if for some reason the images end up being rendered in the viewport with different dimensions, that's where a ratio-box is useful.

In this case, it would actually imply wrapping the whole `picture` element in a relatively positioned `div`, which would have this padding. It's far from clean, but until the [aspect-ratio css property is properly supported][ar], that's [the best we have][sm] on the web. 

[sm]: https://www.smashingmagazine.com/2020/03/setting-height-width-images-important-again/
[ar]: https://caniuse.com/#feat=mdn-css_properties_aspect-ratio

Reply:
This component should ideally be context-agnostic, there's no assumption that it *will* live inside a `picture` element, so in this case you'd have to wrap it directly in a `div`.